internal/commands/pull/inputs.go

```diff
@@ -94,7 +95,18 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 	return app, nil
 }
 
-func (i *inputs) resolveClientTemplates(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+type clientTemplate struct {
+	id     string
+	client *zip.Reader
+}
+
+func (i *inputs) resolveClientTemplates(ui terminal.UI, realmClient realm.Client, groupID, appID string) ([]clientTemplate, error) {
+	if proceed, err := ui.Confirm("Would you like to export with a client template?"); err != nil {
```

Comment:
moved this up so we can avoid making network call if not necessary

Reply:
this should be reverted.  I like the thought process of making things more efficient, but these commands aren't all that expensive to run in the first place.

in general, its nice to avoid prompts where we can (much like you try to minimize a user's button clicks in a UI).  if the user set `--template` in the original command, its safe to assume the answer to this prompt is always "yes".

also, the extra network call here serves as a quick check that we even know the options available to present to the user, if/when they _do_ select "yes" when wishing to continue